src/s1reader/s1_reader.py

```diff
@@ -526,6 +529,100 @@ def get_track_burst_num(track_burst_num_file: str = esa_track_burst_id_file):
 
     return track_burst_num
 
+
+def get_ascending_node_time_orbit(orbit_state_vector_list: ET,
+                                  sensing_time: datetime.datetime,
+                                  anx_time_annotation: datetime.datetime=None):
+    '''
+    Estimate the time of ascending node crossing from orbit
+
+    Parameters
+    ----------
+    orbit_state_vector_list: ET
+        XML elements that points to the list of orbit information.
+        Each element should contain the information below:
+        TAI, UTC, UT1, Absolute_Orbit, X, Y, Z, VX, VY, VZ, and Quality
+
+    sensing_time: datetime.datetime
+        Sensing time of the data
+
+    anx_time_annotation: datetime.datetime
+        ascending node crossing (ANX) time retrieved from annotation data.
+        - If it is provided, then the orbit ANX time close to this will be returned.
+        - If it is `None`, then the orbit ANX time closest to (but no later than)
+          the sensing time will be returned.
+
+    Returns
+    -------
+    _ : datetime.datetime
+        Ascending node crossing time calculated from orbit
+    '''
+
+    # Crop the orbit information before 2 * (orbit period) of sensing time
+    search_length = datetime.timedelta(seconds=2 * T_ORBIT)
+    orbit_until_sensing_time = get_burst_orbit(sensing_time  - search_length,
```

Comment:
I think there's a way we could simplify this function by making `get_burst_orbit` a little more general (maybe make its hardcoded `pad_time` a function argument so you don't have to do the manual padding here). but not worth delaying this PR more